src/mlpack/methods/cf/neighbor_search_policies/pearson_search.hpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file pearson_search.hpp
+ * @author Wenhao Huang
+ *
+ * Nearest neighbor search with pearson distance.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_CF_PEARSON_SEARCH_HPP
+#define MLPACK_METHODS_CF_PEARSON_SEARCH_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/methods/neighbor_search/neighbor_search.hpp>
+
+namespace mlpack {
+namespace cf {
+
+/**
+ * Nearest neighbor search with pearson distance (or furthest neighbor search
+ * with pearson correlation).
+ * Note that, with normalized vectors, neighbor search with pearson distance is
+ * equivalent to neighbor search with Euclidean distance. Therefore, instead
+ * of performing neighbor search directly with pearson distance, we first
+ * normalize all vectors, and then use neighbor::KNN (i.e. NeighborSearch with
+ * Euclidean distance, KDTree). Pearson correlation are calculated from
+ * Euclidean distance.
+ */
+class PearsonSearch
+{
+ public:
+  /**
+   * Constructor with reference set.
+   * In order to use neighbor::KNN(i.e. NeighborSearch with Euclidean distance,
+   * KDTree), we need to normalize all vectors in referenceSet. For each vector
+   * x, we first subtract mean(x) from each element in x. Then, we normalize
+   * the vector to unit length.
+   *
+   * @param Set of reference points.
+   */
+  PearsonSearch(const arma::mat& referenceSet)
+  {
+    // Normalize all vectors in referenceSet.
+    // For each vector x, first subtract mean(x) from each element in x.
+    // Then normalize the vector to unit length.
+    arma::mat normalizedSet(arma::size(referenceSet));
```

Comment:
Can we use ``arma:: normalise`` here?

Reply:
Yes we should use `normalise()` instead.